vpr/src/place/RL_agent_util.cpp

```diff
@@ -72,7 +117,14 @@ void create_move_generators(std::unique_ptr<MoveGenerator>& move_generator, std:
                 move_generator2 = std::make_unique<SimpleRLMoveGenerator>(karmed_bandit_agent2);
             } else {
                 //agent's 1st state
-                karmed_bandit_agent1 = std::make_unique<SoftmaxAgent>(NUM_PL_NONTIMING_MOVE_TYPES);
+                if (placer_opts.place_agent_space == e_agent_space::MOVE_BLOCK_TYPE) {
+                    VTR_LOG("Using simple RL 'Softmax agent' for choosing move and block types\n");
+                    karmed_bandit_agent1 = std::make_unique<SoftmaxAgent>(NUM_PL_NONTIMING_MOVE_TYPES,
+                                                                          place_ctx.agent_to_logical_map.size());
+                } else {
+                    VTR_LOG("Using simple RL 'Softmax agent' for choosing move types\n");
+                    karmed_bandit_agent1 = std::make_unique<SoftmaxAgent>(NUM_PL_NONTIMING_MOVE_TYPES);
+                }
```

Comment:
Take a look and see if some ifs can be collapsed with helper functions or variables you set and use later.